Added authentication flow and action queues

nraomobiquityinc opened the pull request:
1. Added Slacker authentication flow
2. Added example authenticated action for deleting a user's files

nraomobiquityinc committed:
First pass at authenticated actions:
1. Added Slacker authentication flow
2. Added example authenticated action for deleting a user's files

nraomobiquityinc commented:
@Ustice Once we've reviewed this, I'll update the README with details about starting redis etc.

nraomobiquityinc committed:
First pass at #2, #3:
* Added Express 4
* All routing code is now centralized in router.js

nraomobiquityinc committed:
Merging ustice/Slacker.

nraomobiquityinc committed:
Fixing package.json typo.

nraomobiquityinc committed:
Fixes #5. Added a guard to meme.js

nraomobiquityinc committed:
Fixed typo in weather.js

nraomobiquityinc committed:
Updating description for deleteFiles action.

nraomobiquityinc committed:
Waypoint towards closing #3:
* added queues for pending actions needing authorization
* updating authentication flow for queues (this has not been completed
  as yet)
* added Handlebar landing page shown to user post authentication

nraomobiquityinc committed:
Finished preliminary validation of action queue...
... workflow. Needs code review!

nraomobiquityinc committed:
Updating contributors in package.json

Ustice commented on actions/deleteFiles.js:
```diff
@@ -0,0 +1,67 @@
+    var accessToken = data.accessToken;
+    listFiles(deleteFile);
+
+    function listFiles(deleteFile) {
```
Move outside of the action.

Ustice commented on authenticationHelper.js:
```diff
@@ -0,0 +1,251 @@
+        //enqueue this action and give him another chance to authorize us
+        pastQueuedActionsList = JSON.parse(pastQueuedActionsList);
+        pastQueuedActionsList.push(actionData);
+        redisClient.hset(data.user_id, "queuedActions", JSON.stringify(pastQueuedActionsList), function(err, res) {
```
Wrap reddis calls to keep code DRY.

Ustice commented on authenticationHelper.js:
```diff
@@ -0,0 +1,251 @@
+            });
+          })
+        }
+      })
```
semicolon

nraomobiquityinc committed:
Changes based on pull review comments:
* increasing code reuse in authenticationHelper.js
* removed obsolete field trigger_word in bot.js
* moving functions that should be local to deleteFiles.js

nraomobiquityinc committed:
First step towards adding promises:
* Added slack-api to deleteFiles.js

nraomobiquityinc committed:
Adding promises to deleteFiles.js. Once the slack-api library is updated
to add promisifyAll, I'll update the code to use it.

nraomobiquityinc committed:
Updating deleteFiles to use new slack-api promisify()

nraomobiquityinc committed:
Improvements to router.js:
* Changed middleware to log requests on all routes
* Refactored the code to make it cleaner
* Changed the runactions route to agree with RESTful resource guidelines

nraomobiquityinc committed:
Updating authSuccess.hbs for the new runactions path.

nraomobiquityinc committed:
Updating deleteFiles.js to add catch for promisify now that slack-api throws out errors correctly.

nraomobiquityinc committed:
Updated bot.sendMessage and added a callback...
... to the code that calls it from authenticationHelper.js so we know if
there's any failure.

nraomobiquityinc committed:
Updated name of delete files command from 'rm' to 'deleteFiles'.

nraomobiquityinc committed:
Added error handling for when Slack returns a 500 response for the
users.info endpoint.

nraomobiquityinc committed:
Added variable to check if argument is undefined.
Bumping slack-api version to 0.10.11 for promisify bug fixes.

nraomobiquityinc commented:
Going to close this request and make a pull to another branch so we don't pollute this timeline.
